File: src/etch/bindings/java/msg/Validator_StructValue.java
/* $Id$
 *
 * Created by wert on Aug 27, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for StructValue.
 */
public class Validator_StructValue extends TypeValidator
{
	/**
	 * @return an instance of the validator.
	 */
	public static Validator_StructValue get()
	{
		if (validator == null)
			validator = new Validator_StructValue();
		
		return validator;
	}
	
	private static Validator_StructValue validator;
	
	/**
	 * Constructs the Validator.
	 */
	private Validator_StructValue()
	{
		super( StructValue.class, StructValue.class, 0, "StructValue" );
	}

	@Override
	public Validator elementValidator()
	{
		return null;
	}
	
	@Override
	public Byte checkValue( Object value )
	{
		if (!validate( value))
			return null;
		
		return TypeCode.STRUCT;
	}
}

File: src/etch/bindings/csharp/msg/Validator_StructValue.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for structvalue
    /// </summary>
    public class Validator_StructValue : TypeValidator
    {
        public static Validator_StructValue Get(XType type, int nDims)
        {
            CheckDims(nDims);          
            return new Validator_StructValue(type,nDims);
        }

       

        private Validator_StructValue(XType type, int nDims)
            : base( typeof( StructValue ), typeof( StructValue ), nDims, string.Format("StructValue[{0},{1}]",type,nDims) )
        {
            this.type = type;
        }

        private XType type;

        public XType GetXType()
        {
            return type;
        }

        public override Validator ElementValidator()
        {
            return Get(type, nDims - 1);
        }

        public override bool Validate( object value )
        {
            return
                base.Validate(value) &&
                (value.GetType() != typeof (StructValue) || type.IsAssignableFrom(((StructValue) value).GetXType));
 
        }     
    }
}
